src/main/java/com/google/sps/servlets/ProfileBuilderServlet.java

```diff
@@ -0,0 +1,66 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.ProfessionalProfile;
+import com.google.sps.data.StudentProfile;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/profile")
+public class ProfileBuilderServlet extends HttpServlet {
+
+  private void storeProfile(String userType, String fullname, String input2, String input3) {
+
+    if(userType.contains("student")){
+
+        Entity profileEntity = new Entity("StudentProfile");
+        profileEntity.setProperty("fullname", fullname);
+        profileEntity.setProperty("school", input2);
+        profileEntity.setProperty("major", input3);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.put(profileEntity);
+    }
+    else{
+        Entity profileEntity = new Entity("ProfessionalProfile");
+        profileEntity.setProperty("fullname", fullname);
+        profileEntity.setProperty("company", input2);
+        profileEntity.setProperty("careerTitle", input3);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.put(profileEntity);
+    }
+  }
+
+
+ @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String profile = request.getParameter("profile-section");
+    String fullname = request.getParameter("name-input");
+    String userType = request.getParameter("user-Type");
+    String school = request.getParameter("school-input");
+    String major = request.getParameter("major-input");
+    String company = request.getParameter("company-input");
+    String careerTitle = request.getParameter("careerTitle-input");
+    String profName = request.getParameter("profName-input");
+
+    if(userType.contains("student")){
+        storeProfile(userType, fullname, school, major);
+    }
+    else{
```

Comment:
Could you explicitly check for "professional" here as an else-if (L60), and then throw or log an error if userType isn't one of the two (in the final else branch)? I get that these are the only two options the user can put in the UI, but in general, it's better to explicitly check and generate a sensible error that can in future quickly identify future problems related to this (e.g.: if someone performing maintenance accidentally leavs userType empty -- in the present logic this would create a professional profile, which is non-intuitive and not helpful to future debugging).

Reply:
done